shims/spark301/src/main/scala/com/nvidia/spark/rapids/shims/spark301/SparkBaseShims.scala

```diff
@@ -652,4 +652,11 @@ abstract class SparkBaseShims extends SparkShims {
   override def hasAliasQuoteFix: Boolean = false
 
   override def hasCastFloatTimestampUpcast: Boolean = false
+
+  override def toGpuExpression(exp: Expression): Expression = {
```

Comment:
The name of this method is super confusing.  The name implies it will take an expression and convert it to a GPU expression, but then apparently only does it for the very specific `UnaryMinus` expression and even then doesn't recursively try to translate any of the child expressions.  Either the name is too generic or this doesn't do what we need it to do.

Reply:
I anticipated there will be more of this kind, and thought of using a single template for all of them instead of adding specific methods all the time

Reply:
this still doesn't do what we want.  The name implies it will convert an expression to a GPU expression, like what GpuOverrides does, but it does not do that at all.  IMO we shouldn't be shimming this in the first place.